terraform/api_gateway.tf.json.template.py

```diff
@@ -652,13 +647,14 @@ def for_domain(cls, domain):
                                         {
                                             'Effect': 'Allow',
                                             'Principal': {
-                                                # REVIEW: Who or what creates the role this ARN is referring to?
-                                                # And what does the part after the last slash in the ARN signify?
                                                 'AWS': 'arn:aws:sts::'
                                                        + aws.account
                                                        + ':assumed-role/'
-                                                       + config.qualified_resource_name(app.name) + '/'
-                                                       + config.qualified_resource_name(app.name, '-notify')
+                                                       + '${aws_lambda_function.' + app.name + '.function_name}/'
+                                                       # The following is the role-session-name of the principal
+                                                       # assuming the role via an AWS STS AssumeRole operation.
+                                                       + '${aws_lambda_function.' + '_'.join([app.name, 'notify'])
+                                                       + '.function_name}'
```

Comment:
[Assumed-role session principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_principal.html#principal-role-session)

Reply:
This diverges from convention, add policy statement to appropriate Lambda role policy.